lib/Lift.cpp

```diff
@@ -590,21 +590,39 @@ bool LiftCodeIntoModule(const remill::Arch *arch, const Program &program,
 
   // Lift global variables.
   program.ForEachVariable([&](const anvill::GlobalVarDecl *decl) {
-    const auto addr = decl->address;
-    const auto name = anvill::CreateVariableName(addr);
-    const auto gvar = decl->DeclareInModule(name, module);
+    const auto addr{decl->address};
+    const auto name{anvill::CreateVariableName(addr)};
+    const auto gvar{decl->DeclareInModule(name, module)};
 
+    // Check if we have mapped bytes
+    if (!program.FindByte(addr)) {
+      return true;
+    }
     // Set initializer
-    auto init = CreateConstFromMemory(addr, decl->type, arch, program, module);
+    auto init{CreateConstFromMemory(addr, decl->type, arch, program, module)};
     gvar->setInitializer(init);
-
     return true;
   });
 
   // Lift functions.
   program.ForEachFunction([&](const FunctionDecl *decl) {
-    const auto entry = lifter.LiftFunction(*decl);
-    DefineNativeToLiftedWrapper(arch, *decl, entry);
+    // Initialize function entry
+    const auto name{CreateFunctionName(decl->address)};
+    FunctionEntry entry{nullptr, nullptr, nullptr};
+    entry.lifted_to_native =
+        remill::DeclareLiftedFunction(&module, name + ".lifted_to_native");
+    entry.native_to_lifted = decl->DeclareInModule(name, module, true);
+
+    // Check if we have mapped bytes
+    if (auto start{program.FindByte(decl->address)};
+        start && start.IsExecutable()) {
+      entry = lifter.LiftFunction(*decl);
+      DefineNativeToLiftedWrapper(arch, *decl, entry);
+    } else {
```

Comment:
Is it worthwhile to consider the case where data exists, but `false == start.IsExecutable()` ?

Reply:
Do the comments near [writeable](https://github.com/lifting-bits/anvill/blob/561e2f3ffda9b81de6878502946e3aa1791fd4bf/lib/Program.cpp#L70) and [executable](https://github.com/lifting-bits/anvill/blob/561e2f3ffda9b81de6878502946e3aa1791fd4bf/lib/Program.cpp#L77) byte metada answer the question?

IIRC this information is derived from the JSON spec so whatever the IDA or Binja frontend tells us is going to decide.